peps/pep-0728.rst

```diff
@@ -554,42 +627,68 @@ Because it did not offer a way to specify the type of the extra items, the type
 checkers will need to assume that the type of the extra items is ``Any``, which
 compromises type safety. Furthermore, the current behavior of TypedDict already
 allows untyped extra items to be present in runtime, due to structural
-subtyping.
+subtyping. ``closed=True`` plays a similar role in the current proposal.
 
 Supporting ``TypedDict(extra=type)``
 ------------------------------------
 
-This adds more corner cases to determine whether a type should be treated as a
-type or a value. And it will require more work to support using special forms to
-type the extra items.
-
-While this saves us from reserving an attribute for special use, it will require
-extra work to implement inheritance, and it is less natural to integrate with
-generic TypedDicts.
+While this design is potentially workable, there are several partially
+addressable concerns to consider. Adding everything up, it is slightly less
+favorable than the current proposal.
+
+- Usability of forward reference
+  As in the functional syntax, using a quoted type or a type alias will be
+  required when SomeType is a forward reference. This is already a requirement
+  for the functional syntax, so implementations can potentially reuse that piece
+  of logic, but this is still extra work that the ``closed=True`` proposal doesn't
+  have.
+
+- Concerns about using type as a value
+  Whatever is not allowed as the value type in the functional syntax should not
+  be allowed as the argument for extra either. While type checkers might be able
+  to reuse this check, it still needs to be somehow special-cased.
+
+- Support for the functional syntax
```

Comment:
I don't understand this point. Seems to me like the `extra=SomeType` form would work great with the functional form.

Reply:
I was trying to argue that adding `closed=True` would be easier to support compared to `extra=SomeType`.

Now that I discovered the deprecated keyword arg syntax for defining TypedDict keys, in the end, this form probably doesn't require much additional work to support.

Reply:
I am pretty sure it would be easier to support the `extra=SomeType` form at runtime than the current proposal.

Reply:
> deprecated keyword arg syntax for defining TypedDict keys

This is a new one on me. Does this not cause backward compatibility issues with *both* keyword proposals?

Reply:
The keyword arg alternative is planned to be removed in 3.13, so by the time PEP 728 can be implemented, that alternative will be gone. We can mention this detail in the backwards compatibility section.

Reply:
I removed the `Support for the functional syntax` part in a later commit.

Reply:
@alicederyn it sort of does for typing-extensions, yes, but I think I can live with that, especially since under the new proposal the only valid values for `closed` are True/False, which aren't valid types anyway.